Update mapbox gl and react-scripts

clinct opened the pull request:
Also:

* Simplifies proxying by no longer requiring a `server.js` in development
* Cleans up some js-cde
* Adds [Netlify _redirects file](https://www.netlify.com/docs/redirects/#proxying)

clinct committed:
implement changes to update

clinct committed:
update to react-scripts 1.1.4 and set direct proxy

clinct committed:
delete unnecessary server.js

clinct committed:
simplify refactor js

clinct committed:
remove wrongly checked in files

clinct committed:
update Dockerfile

clinct committed:
add _redirects for netlify proxy

clinct committed:
update readme

clinct committed:
update README

clinct committed:
add point-input field

clinct committed:
Correctly add node_modules

clinct committed:
allow to add multiple latlons

comlaterra commented on README.md:
```diff
@@ -24,37 +24,19 @@ App that uses travel-matrix to get the best route between points which has been
-
-Starting FE
-* cd IN to travel-matrix-react-map folder
 * `npm install`
```
* cd `app`

comlaterra commented on app/package.json:
```diff
@@ -3,23 +3,28 @@
+      "target": "http://dev05.c42.io:5003"
+    }
+  }
 }
```
The dependencies 
* express
* immutable
* react-bootstrap

Seems to be unnecessary after this merge.
If so we can just remove them

comlaterra commented on app/src/MatrixService.js:
```diff
@@ -1,16 +1,19 @@
   }
-  // TODO: construct this URL in a nicer way
-  return fetch((process.env.REACT_APP_MATRIX_URL ? process.env.REACT_APP_MATRIX_URL : '') + "/proxy/matrix/?departures="+arrivals+"&arrivals="+departures+"&debug=true&search_range=100")
+
```
Why to remove the TODO? :P

clinct replied:
Because it got nicer? Although, only slightly :)

comlaterra commented on app/src/GeoJSONMap.js:
```diff
@@ -1,57 +1,42 @@
+  };
 
-  newMarker = function (lngLat) {
+  _newMarker = function (lngLat) {
```
What's the point on using underscore convention?
P.S: JS allows encapsulation, if desired. Through the contructor pattern either in [JS](http://www.samselikoff.com/blog/some-Javascript-constructor-patterns/) or [ES](https://medium.com/javascript-scene/javascript-factory-functions-vs-constructor-functions-vs-classes-2f22ceddf33e)

clinct replied:
For me it marks methods that are not directly called from JSX. Would encapsulation work for that for you?

comlaterra replied:
I though you were using the common underscore convention for "internal" methods in Python.

comlaterra reviewed (commented):
* README MarkdownTOC not working
* Some other comments.

I'll keep it up to clean up all PR's in FE projects.

clinct committed:
update TOC README

clinct committed:
update README

clinct committed:
removed server.js related dependencies

clinct commented:
Hi @comlaterra !

I updated the README and removed server.js dependencies. Also see other small comments above.

I btw left react-bootstrap for now, will probably be used soon ;)

Thanks for reviewing!

clinct committed:
Use dataDrivencolorStyle and react-bootstrap

clinct committed:
async handling of onSubmittedPoint

clinct committed:
add speed property to GeoJSON

comlaterra commented on app/src/MatrixService.js:
```diff
@@ -34,10 +34,21 @@ export function getGeoJson (departures, arrivals) {
+                    } else if (f["properties"]["linker"] === "as_the_crow") {
+                      f["properties"]["speed"] = 0
+                    } else if (f["properties"]["linker"]) {
+                      f["properties"]["speed"] = 10
```
This may deserve a comment and to convert the magic 10 to a descriptive named var (?).
Also, just to know, what is the colour if "speed" is not defined?

clinct replied:
Hehe, indeed quite magicy.. :) The 10 is just to let the linkers be distinguishable from other roads in colors.. It's just a speed that should never occur on normal roads...

If speed is not there, it will default to MapBoxGl black

clinct replied:
Variable name proposal 💃  ?

comlaterra commented:
This is too old.
